### PR TITLE
Add start of Jest unit testing

### PR DESCRIPTION
**Pull request by loneil**
# Description

Add a unit test framework for the frontend application.
Use Jest, through Vue Test Utils (https://vue-test-utils.vuejs.org/). Added with Vue CLI

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation (non-breaking change with enhancements to documentation)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have checked that unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: frontend/package.json [hunk 1]
```diff
@@ -64,14 +67,38 @@
...
   "jest": {
+    "setupFiles": [
+      "<rootDir>/tests/unit/globalSetup.js"
+    ],
     "moduleFileExtensions": [
       "js",
       "json",
-      "vue"
+      "vue",
+      "jsx"
```
Comment (jujaga) [inline frontend/package.json]: Not too sure why we're adding jsx support as that's a React specific extension

File: frontend/package.json [hunk 2]
```diff
@@ -64,14 +67,38 @@
...
     ],
     "transform": {
-      ".*\\.(vue)$": "vue-jest"
-    }
+      ".*\\.(vue)$": "vue-jest",
+      "^.+\\.vue$": "vue-jest",
+      ".+\\.(css|styl|less|sass|scss|svg|png|jpg|ttf|woff|woff2)$": "jest-transform-stub",
+      "^.+\\.jsx?$": "babel-jest"
```
Comment (jujaga) [inline frontend/package.json]: Not too sure why we're adding jsx support as that's a React specific extension

File: frontend/package.json [hunk 3]
```diff
@@ -64,14 +67,38 @@
...
+      "/node_modules/"
+    ],
+    "moduleNameMapper": {
+      "^@/(.*)$": "<rootDir>/src/$1"
+    },
+    "snapshotSerializers": [
+      "jest-serializer-vue"
+    ],
+    "testMatch": [
+      "**/tests/unit/**/*.spec.(js|jsx|ts|tsx)|**/__tests__/*.(js|jsx|ts|tsx)"
```
Comment (jujaga) [inline frontend/package.json]: Not too sure why we're adding jsx support as that's a React specific extension

File: frontend/tests/unit/.eslintrc.js [hunk 4]
```diff
@@ -0,0 +1,5 @@
+module.exports = {
+  env: {
+    jest: true
+  }
+}
```
Comment (jujaga) [inline frontend/tests/unit/.eslintrc.js]: Let's try to have a newline at the end of the file if possible

File: frontend/tests/unit/globalSetup.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+const logError = console.error; // eslint-disable-line no-console
+console.error = (...args) => { // eslint-disable-line no-console
+    if (
+        args[0].includes("[Vuetify]") &&
+        args[0].includes("https://github.com/vuetifyjs/vuetify/issues/4068")
+    )
+        return;
+    logError(...args);
+};
```
Comment (jujaga) [inline frontend/tests/unit/globalSetup.js]: Let's try to have a newline at the end of the file if possible

File: frontend/tests/unit/components/Footer.spec.js [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+  let wrapper;
+
+  beforeEach(() => {
+    const localVue = createLocalVue();
+
+    localVue.use(Vuetify);
+
+    wrapper = mount(Footer, {
+      localVue: localVue
+
```
Comment (jujaga) [inline frontend/tests/unit/components/Footer.spec.js]: extra blank line

File: frontend/package.json (hunk 1 above)
Comment (loneil) [reply frontend/package.json]: Added by Vue CLI Jest scaffolding (for some reason...). Removing manually then re-running vue add adds it back, so we'll leave it.

File: frontend/package.json (hunk 3 above)
Comment (loneil) [reply frontend/package.json]: Added by Vue CLI Jest scaffolding (for some reason...). Removing manually then re-running vue add adds it back, so we'll leave it.

File: frontend/tests/unit/globalSetup.js (hunk 5 above)
Comment (loneil) [reply frontend/tests/unit/globalSetup.js]: Fixed

File: frontend/tests/unit/components/Footer.spec.js (hunk 6 above)
Comment (loneil) [reply frontend/tests/unit/components/Footer.spec.js]: Fixed

**Pull request merged**